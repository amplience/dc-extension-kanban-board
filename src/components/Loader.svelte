<script lang="ts">
  const dots = new Array(3);
</script>

<style lang="scss">
  .spinner-container {
    display: flex;
    justify-content: center;
    align-items: center;
    position: absolute;
    z-index: 10;
    width: 100%;
    height: 100%;
    opacity: 0.8;
    transition: opacity 0.3s;
    background-color: #fff;

    .spinner {
      width: 8px;
      height: 8px;
      margin: 1px;
      background-color: #666;
      border-radius: 100%;
      display: inline-block;
      animation: sk-bouncedelay 1.4s infinite ease-in-out both;
    }
    .dot-1 {
      animation-delay: -0.16s !important;
    }
    .dot-2 {
      animation-delay: -0.32s !important;
    }
    @-webkit-keyframes sk-bouncedelay {
      0%,
      80%,
      100% {
        transform: scale(0);
      }
      40% {
        transform: scale(1);
      }
    }
    @keyframes sk-bouncedelay {
      0%,
      80%,
      100% {
        transform: scale(0);
      }
      40% {
        transform: scale(1);
      }
    }
  }
</style>

<div class="spinner-container">
  {#each dots as _, i}
    <div class="spinner dot-{i}" />
  {/each}
</div>
