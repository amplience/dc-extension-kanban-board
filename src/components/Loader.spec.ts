import { render } from '@testing-library/svelte';
import Loader from './Loader.svelte';

describe('loader component', () => {
  it('should render a loader component', () => {
    const { container } = render(Loader, {
      // no-op.
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
