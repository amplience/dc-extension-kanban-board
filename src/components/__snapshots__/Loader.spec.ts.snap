// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader component should render a loader component 1`] = `
<div>
  <div
    class="spinner-container svelte-1rhu03n"
  >
    <div
      class="spinner dot-0 svelte-1rhu03n"
    />
    <div
      class="spinner dot-1 svelte-1rhu03n"
    />
    <div
      class="spinner dot-2 svelte-1rhu03n"
    />
  </div>
</div>
`;
